### PR TITLE
Pasha Business Views Added

### PR DESCRIPTION
**Pull request by faridpashabank**

### CODE REVIEW

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+        }
+    }
+    
+    /// The color of button's border.
+    ///
+    /// By default button will be created with the border color for selected button style.
+    ///
+    public var borderColor: UIColor = .clear {
+        didSet {
+            self.baseView.layer.borderWidth = 1
```
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:156]: Why are you updating the `borderWidth` on every `borderColor` change?

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+    /// * Title color
+    /// * Tint color
+    ///
+    public var theme: PBBUIButtonTheme = .regular {
+        didSet {}
+    }
+    
+    private var typeOfAction: PBBActionType = .normal(localizedTitleText: "") {
+        didSet {}
+    }
```
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:175]: Empty `didSet`s

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift [hunk 3]
```diff
@@ -0,0 +1,710 @@
...
+    
+    public var statusTypeOfAction: PBBLabelView.PBBLabelViewStatus = .new {
+        didSet {}
+    }
+    
+    public var styleOfAction: PBBActionStyle = .none {
+        didSet {}
+    }
```
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:191]: Is there any specific reason that you kept them empty?

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift [hunk 4]
```diff
@@ -0,0 +1,710 @@
...
+    }()
+    
+    override private init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        fatalError("init(coder:) has not been implemented")
+    }
```
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:383]: I think if you are not doing anything on the init call, there is not need to override them

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift [hunk 5]
```diff
@@ -0,0 +1,710 @@
...
+
+    /// Creates a new button of specified style.
+    ///
+    /// - Parameters:
+    ///    - localizableTitle: Sets the title text for button.
+    ///    - typeOfButton: Sets the type of button.
+    ///
+    
+    public convenience init(typeOfAction: PBBActionType = .normal(icon: .none, localizedTitleText: "")) {
+        self.init()
```
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:393]: Also applies to these `self.init()` callls

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift [hunk 6]
```diff
@@ -0,0 +1,710 @@
...
+//            self.titleLabel.font = UIFont.sfProText(ofSize: 13, weight: self.buttonTitleWeight)
+//            self.iconBackgroundColor = self.theme.getPrimaryColor()
+//            self.iconWrapperView.backgroundColor = self.theme.getPrimaryColor()
+        case .disabled: break
+//            self.titleLabel.font = UIFont.sfProText(ofSize: 13, weight: .semibold)
+//            self.titleLabel.textColor = UIColor.Colors.PBBGray
+//            self.disableTitleLabel.font = UIFont.sfProText(ofSize: 11, weight: .medium)
+//            self.disableTitleLabel.textColor = .white
+//            self.iconBackgroundColor = UIColor.Colors.PBBBackgroundGray
+        case .selected: break
```
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:649]: If you will refer back to this part, please add TO-DOs

File: Sources/PashaKit/PBBusiness/Buttons/PBBRoundButton.swift
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/Buttons/PBBRoundButton.swift:1]: Fully commented out class

File: Sources/PashaKit/PBBusiness/Buttons/PBBUIButton.swift [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+    }
+
+    override private init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        fatalError("init(coder:) has not been implemented")
+    }
```
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/Buttons/PBBUIButton.swift:207]: Unused `init` overrides

File: Sources/PashaKit/PBBusiness/TextFields/PBBTextField.swift
Comment (muradtries) [inline Sources/PashaKit/PBBusiness/TextFields/PBBTextField.swift:1]: If nearly anything is identical by design, maybe you can create subclass of `PBUITextField` and extend it with your enums

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift (hunk 4 above)
Comment (faridpashabank) [reply Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:383]: removed

File: Sources/PashaKit/PBBusiness/Buttons/PBBRoundButton.swift
Comment (faridpashabank) [reply Sources/PashaKit/PBBusiness/Buttons/PBBRoundButton.swift:1]: What you mean?

File: Sources/PashaKit/PBBusiness/Buttons/PBBUIButton.swift (hunk 7 above)
Comment (faridpashabank) [reply Sources/PashaKit/PBBusiness/Buttons/PBBUIButton.swift:207]: removed

File: Sources/PashaKit/PBBusiness/TextFields/PBBTextField.swift
Comment (faridpashabank) [reply Sources/PashaKit/PBBusiness/TextFields/PBBTextField.swift:1]: I don't want to dependance

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift (hunk 6 above)
Comment (faridpashabank) [reply Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift:649]: Added

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift (hunk 1 above)
Comment (RamazanRustamov) [reply Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift]: I agree with Murad, let's leave freedom to choose border width to developer. I think if you're indicating border color, you should know that setting border width as well is a must.

File: Sources/PashaKit/PashaBusiness/Buttons/SMEUIButton.swift [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+
+    /// The tint color to apply to the button title and image.
+    ///
+    /// By default button will be created with the tint color for selected button style.
+    ///
+    public var buttonTintColor: UIColor = UIColor.white {
+        didSet {
+            self.tintColor = self.buttonTintColor
+        }
+    }
```
Comment (RamazanRustamov) [inline Sources/PashaKit/PashaBusiness/Buttons/SMEUIButton.swift:143]: [Optional] I would suggest changing properties of Foundation and UIKit frameworks to simple `get {}` and `set {}`, as observing functionality is not doing anything, just sets certain property.

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift (hunk 5 above)
Comment (faridpashabank) [reply Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift]: Apply what? Remove self.init()?

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift (hunk 3 above)
Comment (faridpashabank) [reply Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift]: I have added TODO: for return it again

File: Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift (hunk 5 above)
Comment (muradtries) [reply Sources/PashaKit/PBBusiness/ActionViews/PBBActionView.swift]: Yes

**Review by muradtries — APPROVED** [review]

**Review by asadovaa — APPROVED** [review]

**Pull request merged**